api/lib/domain/models/CertificationCpfCountry.js

```diff
@@ -0,0 +1,15 @@
+class CertificationCpfCountry {
+  constructor({
+    code,
+    commonName,
+    originalName,
+  }) {
+    this.code = code;
+    this.commonName = commonName;
```

Comment:
Je ne suis pas sûre que vous ayez besoin du commonName, mais là je me trompe peut-être

Reply:
On s'est dit que si le menu était cherchable, on pourrait chercher dans plusieurs graphies différentes et donc avoir les deux côtés front.

Reply:
Est-ce que le menu est effectivement cherchable aujourd'hui, ou est-il prévu qu'il le devienne dans un futur proche (i.e. les prochains tickets de l'epix qui sont dans "ready for dev" ?).

Si c'est pas le cas, soit il faut proposer/faire valider qu'on rend le menu cherchable et on peut le laisser. Dans le cas contraire : on le retire (YAGNI) et on le remettra quand on rendra effectivement le menu cherchable.

Reply:
je crois savoir que la drop down dans PixUi est cherchable, à vérifier.
+ Je renvoie à mon commentaire plus bas, d'avoir toutes les variations, ça rend difficile la navigation dans la drop down. 
Typiquement : aujourd'hui pour le Danemark il existe les valeurs : Danemark et Royaume du Danemark.
Donc si moi, ETQ utilisateur, je commence à saisir Danemark dans la recherche, je vais voir les deux, et je ne saurais pas lequel choisir

Reply:
Je crois que la proposition de @iwazaru c'est de pouvoir matcher sur les deux termes (originalName + commonName) mais de ne remonter dans les suggestions seulement les commonName. Je sais pas si j'ai bien compris ton inquiétude ni bien répondu ^^

Reply:
Donc en gros si tu cherches Danemark et que tu te dis dans ta tête "en fait c'est royaume du Danemark", quand tu commences à taper "Royau" t'as "Danemark" qui sort. Là où si tu matches seulement sur le commonName si ça se trouve en chechant "Royaume du Danemark", "Danemark" ne sortirait même pas.

Reply:
Si je tape seulement "Danemark" j'ai seulement "Danemark" qui sort même si en arrière plan j'ai deux matches ("Danemark" et "Royaume du Danemark") mais c'est deux matches correspondent au même commonName donc je propose que celui-là.

Reply:
Je pense qu'il voulait permettre de chercher les deux. En gros, si tu tapes "République" tu as beaucoup de pays qui sortent. Par ailleurs, il reste le cas un peu particulier de la Tchéquie/République Tchèque (perso j'utilise plutôt le deuxième). De toute façon, je préfère que le Model ici possède tous les champs contenus dans la table car ce n'est pas un Read Model et, si on souhaite ne pas tout retourner au front, c'est le boulot tu serializer. Si on a besoin de l'id, du matcher ou du commonName ailleurs dans l'API, on finira inévitablement par les rajouter donc autant le faire tout de suite.

Reply:
Quote : Si on a besoin de l'id, du matcher ou du commonName ailleurs dans l'API, on finira inévitablement par les rajouter donc autant le faire tout de suite.

YAGNI ? on ne sait pas encore 🤷🏿‍♀️

Reply:
De même, ce que tu proposes @EGI-OCTO ça me paraît un poil plus complexe. En mode v0 personnellement je ne choisirais pas ça, car je suis pratiquement certaine que sur PixUi il existe une drop down cherchable toute prête, mais venir overrider ce comportement par un algo un poil plus poussé, je ne sais pas ce que ça coûte

Reply:
y'a du comportement dans les components PIX-UI 😬 ? Faudrait plutôt passer des lambda depuis l'extérieur pour ne pas introduire du couplage entre différentes parties du front via les comportements built-in Pix-UI

Reply:
ben y'a le "comportement" de : je commence à saisir des lettres dans mon champ cherchable et je limite la liste de la dropdown aux items qui matchent. Un truc attendu d'une drop down cherchable non ?

Reply:
J'ai été jété un oeil et c'est le comportement de base HTML pour les `input` avec `datalist`, donc le comportement n'est pas défini dans PIX-UI. 

Le jour où on a des façons de matcher un peu sioux on pourrait imaginer passer une/des lambda.